packages/core/react/use-chat.ts

```diff
@@ -159,10 +159,38 @@ const getStreamedResponse = async (
 
   if (isComplexMode) {
     while (true) {
-      const { done, value } = await reader.read()
+      const { done, value: partialValue } = await reader.read()
       if (done) {
         break
       }
+
+      let value = partialValue
+
+      if (value[value.length - 1] !== 10) {
+        // if the last character is not a newline (ASCII code 10), we need to read more chunks
+        let buffer = new Uint8Array(value.length * 2);
+        buffer.set(value, 0); // copy the first chunk into the buffer
+        let offset = value.length;
+        while (value[value.length - 1] !== 10) {
+          // read more chunks until we get a newline
+          const { done, value: partialValue } = await reader.read();
+          if (done) {
+            break;
```

Comment:
If we the reader is done and we have a buffer, shouldn't we process that buffer?

Reply:
The reader only ever returns done == true if there is no new data at all. Whenever done == true, the partialValue will be undefined.

https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultReader/read